### PR TITLE
BZ1243740 redo: added file about working with HTTP proxies

### PR DESCRIPTION
**Pull request by bfallonf**
@adellape @tnguyen-rh I'll comment on the other PR, but this is a redo of #878 .

### CODE REVIEW

File: admin_guide/http_proxies.adoc [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+...
+----
+====
+
+You can also use the `oc env` command to update an existing deployment
+configuration with a new environment variable:
+
+====
+----
+oc env dc/frontend HTTP_PROXY=http://USER:PASSWORD@IPADDR:PORT
```
Comment (adellape) [inline admin_guide/http_proxies.adoc]: Missing `$` prompt.

Comment (bfallonf) [inline admin_guide/http_proxies.adoc]: Done

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -519,6 +519,7 @@ For example, defining a custom HTTP proxy to be used during build:
 ----
 ====
 
+// tag::using-a-proxy-for-git-cloning-1[]
```
Comment (adellape) [inline dev_guide/builds.adoc]: Probably should move this tag a few lines down to after "Using a Proxy for Git Cloning" title? The way it is now in the http_proxies topic, the rendered build is getting double headings back to back (both "Git Repository Access" and "Using a Proxy for Git Cloning").

File: admin_guide/http_proxies.adoc [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+# systemctl restart openshift-node
+----
+
+[[proxying-docker-pull]]
+
+== Proxying Docker Pull
+
+OpenShift node hosts need to perform push and pull operations to Docker
+registries. If you have a registry that does not need a proxy for nodes to
+access, include the `*NO_PROXY*` parameter with the registry's hostname, the
```
Comment (adellape) [inline admin_guide/http_proxies.adoc]: s/hostname/host name/

File: admin_guide/http_proxies.adoc [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+registry service's IP address, and service name. This blacklists that registry's
+address, leaving the external http proxy as the only option.
```
Comment (adellape) [inline admin_guide/http_proxies.adoc]: s/http/HTTP/

File: admin_guide/http_proxies.adoc [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+HTTPS_PROXY=https://USERNAME:PASSWORD@10.0.0.1:8080/
+NO_PROXY=master.hostname.example.com
+----
+====
+
+[[configuring-default-templates-for-proxies]]
+
+== Configuring Default Templates for Proxies
+
+The example templates available in OpenShift by default do not include settings
```
Comment (adellape) [inline admin_guide/http_proxies.adoc]: Maybe link "example templates" here to `install_config/install/first_steps.html#creating-instantapp-templates`?

File: admin_guide/http_proxies.adoc [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+====
+----
+...
+source:
+ type: Git
+ git:
+  uri: "git://github.com/openshift/ruby-hello-world.git"
+...
+----
+====
```
Comment (adellape) [inline admin_guide/http_proxies.adoc]: Maybe I'm missing something but this doesn't appear to be showing any added proxy settings? As the next sentence suggests, I would expect that like the "Using a Proxy for Git Cloning" process there would be `httpProxy` and `httpsProxy` fields/values shown here. @bparees Can you confirm?

Comment (bparees) [inline admin_guide/http_proxies.adoc]: confirmed.

Comment (tnguyen-rh) [inline admin_guide/http_proxies.adoc]: Small style nit: It would be nice to indent by the same amount as the other examples (two spaces per sub-level).

Comment (adellape) [inline admin_guide/http_proxies.adoc]: +1

File: dev_guide/builds.adoc (hunk 2 above)
Comment (bfallonf) [inline dev_guide/builds.adoc]: I see what you mean. Done.

**Comment by bfallonf** [comment]
Suggestions done.

@tnguyen-rh can you confirm the indenting thing was done as you suggested? 

Any final suggestions?

**Comment by tnguyen-rh** [comment]
@bfallonf LGTM.

**Comment by tnguyen-rh** [comment]
@bfallonf
No more suggestions, but just a comment on YAML syntax: Unless there are single or double quote chars involved, the value of a field need not be quoted.  (Same goes for the name of a field, btw, but those are almost always simple symbols.)  Of course, there is no problem quoting the value:

```YAML
    uri: "git://github.com/openshift/ruby-hello-world.git"
```

and in fact, it's a good (but unfortunately, not bulletproof) defensive strategy when the value includes replaceables:

```YAML
    value: "http://USER:PASSWORD@IPADDR:PORT"
```

It's not bulletproof because *USER* or *PASSWORD* might contain a double quote... :neutral_face: 

Perhaps we should add to the OpenShift docs style guide something about YAML "best practices".

**Comment by adellape** [comment]
@bfallonf LGTM, but two auto-generated diagram files (from the build process) got added into the commit:

architecture/core_concepts/images/docker-diagram.png
architecture/core_concepts/images/docker-diagram.png.cache

We can merge after those are removed. Side note, this has been fixed in a newer version of the build tools (so the images are cleaned up automatically), which Harrison I believe will be updating us on shortly.

**Comment by bfallonf** [comment]
@tnguyen-rh Ah ok. Thanks. They were leftovers from my copy/paste... I deleted them here for consistency.

@adellape removed them too.

If nothing else, then this should be OK to merge.

**Pull request merged**